### PR TITLE
Fixing master branch - docs-examples tests don't run

### LINKED ISSUE #433: docs-examples tests for 4.0.0 never run
### Issue description

this is similar to #431 but adds additional changes needed for Micronaut 4.0.0

### PR DESCRIPTION
**Pull request by wetted**
fixes #433 

This PR cherry picks the fixes from #432 and adds the necessary changes to work with the master branch targeting Micronaut 4.0.0

### CODE REVIEW

**Comment by wetted** [comment]
A kotlin docs-examples test is failing where it doesn't like the Serdable `Snake` class. Looking into it.

File: docs-examples/example-kotlin/build.gradle [hunk 1]
```diff
@@ -26,7 +21,7 @@ test {
 
 compileTestKotlin {
     kotlinOptions {
-        jvmTarget = '17'
+        jvmTarget = '1.8'
```
Comment (timyates) [inline docs-examples/example-kotlin/build.gradle:24]: Why 1.8?

Comment (wetted) [reply docs-examples/example-kotlin/build.gradle]: That was not intended

**Comment by sdelamo** [comment]
@wetted can you check the failure?

**Comment by wetted** [comment]
> @wetted can you check the failure?

@sdelamo I'm not sure what's going on yet. It's failing on master too (as you can see in the module checker), confirmed by a clean local build. I'll look into it on Monday when I'm back.

**Comment by timyates** [comment]
I have no idea why the messages seem to be lost in the tests 😞

**Comment by wetted** [comment]
ping @sdelamo this is passing now and ready to review/merge, including updates we want for Micronaut 4.0.0.M1

**Comment by timyates** [comment]
@sdelamo @wetted I *think* that's it...

I [had to fix this test](https://github.com/micronaut-projects/micronaut-rabbitmq/pull/434/commits/e83703c686656ac406e17e590c320bb5914fe629) which I think is correct now

Could do with a sanity check though...

I think Sonar is lying about coverage (at least, both I and IntelliJ think it's wrong)

**Comment by wetted** [comment]
> I think Sonar is lying about coverage (at least, both I and IntelliJ think it's wrong)

I have the same issue with a Pulsar PR. We (contributor and I) believe the 0% Coverage failure is bogus.

**Pull request merged**